### DISCUSSION TITLE
Rebuilding Pretix  Requires More Than 2GB Memory It Seems.

### DISCUSSION THREAD

**Discussion by HansF** [pretix/pretix#3422] [Q&A]
I've been running Pretix on 2GB memory Digital Ocean servers for several years without any issues. However, I recently encountered a problem with the python -m pretix rebuild command, which consistently gets killed before completion when run on a server with this amount of memory.

I attempted the rebuild process multiple times, ensuring that the server had as much free resources as possible by not running anything else. Despite these attempts, the command continued to be killed before it could complete, it displayed the 'static files' line but never the "Compressing... done" .

Upon upgrading the server to 4GB of memory, the rebuild process completed smoothly without any issues. This suggests that the process may now require more than 2GB of memory to complete, although I had not encountered this requirement in the past.

I've checked the official Pretix documentation for any updates on system requirements, but I didn't find any specific details on memory needs. I plan on keeping my server at 4GB for now to prevent further issues.

Can you provide some insight on the memory requirements for the rebuild process?

**Comment by raphaelm** [comment]
Hmm. Yes, we could document system requirements somewhere, but then we'd probably need to have some way of testing if the numbers hold up over time?

**Comment by HansF** [comment]
Good point, thanks for your answer.

**Comment by ensko** [comment]
I ended up installing the same version locally, running `rebuild`, and then rsyncing the result over to my host. As far as I can tell, this seems to have worked.
